snuba/query/allocation_policies/per_referrer.py

```diff
@@ -100,6 +117,18 @@ def _get_quota_allowance(
             query_id,
             rate_limit_params,
         )
+        assert (
+            rate_limit_params.concurrent_limit is not None
+        ), "concurrent_limit must be set"
+        num_threads = self._get_max_threads(referrer)
+        requests_throttle_threshold = self.get_config_value(
+            "default_concurrent_request_per_referrer"
+        ) // self.get_config_value("requests_throttle_divider")
+        if rate_limit_stats.concurrent > requests_throttle_threshold:
+            num_threads //= self.get_config_value("threads_throttle_divider")
```

Comment:
I think it's less clean to mutate this variable, as opposed to setting it [once] conditionally

Additionally, if your divider is set to 2 and your max_threads is 1, then I think this logic might result in an error (`1 // 2` evaluates to 0 in python). However we want to handle this, there should be a test for it, or a guardrail that prevents the bad config from being used in production

Reply:
I might suggest for simplicity that you just have a static throttled_thread_count

Reply:
I think the intention is to auto-adjust the warning or throttling threshold when the rejection threshold is adjusted (ie prevents human error like setting rejection threshold to 20 but warning threshold was left at 40). I used `max` to ensure we will always execute with at least 1 thread

Reply:
@onewland The static threshold becomes annoying in the case of overrides. Every time you override a referrer's concurrent queries you would also need to override it's throttle threshold.